packages/office-ui-fabric-react/src/utilities/DraggableZone/DraggableZone.tsx

```diff
@@ -0,0 +1,340 @@
+    ) {
+      return;
+    }
+
+    // Remember the touch identifier if this is a touch event so we can
+    // distinguish between individual touches in multitouch scenarios
+    // by remembering which touch point we were given
+    this._touchId = this._getTouchId(event);
+
+    const position = this._getControlPosition(event);
+    if (position === undefined) {
+      return;
+    }
+
+    const dragData = this._createDragDataFromPosition(position);
+    this.props.onStart && this.props.onStart(event, dragData);
+
+    this.setState({
+      isDragging: true,
+      lastPosition: position
+    });
+
+    // hook up the appropriate mouse/touch events to the body to ensure
+    // smooth dragging
+    this._events.on(document, this._currentEventType.move, this._onDrag);
+    this._events.on(document, this._currentEventType.stop, this._onDragStop);
+  };
+
+  private _onDrag = (event: MouseTouchEvent<HTMLElement>) => {
+    // Prevent scrolling on mobile devices
+    if (event.type === 'touchmove') {
+      event.preventDefault();
+    }
+
+    const position = this._getControlPosition(event);
+    if (!position) {
+      return;
+    }
+
+    // create the updated drag data from the position data
+    const updatedData = this._createUpdatedDragData(this._createDragDataFromPosition(position));
+    const updatedPosition = updatedData.position;
+
+    this.props.onDragChange && this.props.onDragChange(event, updatedData);
+
+    this.setState({
+      position: updatedPosition,
+      lastPosition: position
+    });
+  };
+
+  private _onDragStop = (event: MouseTouchEvent<HTMLElement>) => {
+    if (!this.state.isDragging) {
+      return;
+    }
+
+    const position = this._getControlPosition(event);
+    if (!position) {
+      return;
+    }
+
+    const baseDragData = this._createDragDataFromPosition(position);
+
+    // Set dragging to false and reset the lastPosition
+    this.setState({
+      isDragging: false,
+      lastPosition: undefined
+    });
+
+    this.props.onStop && this.props.onStop(event, baseDragData);
+
+    if (this.props.position) {
+      this.setState({
+        position: this.props.position
+      });
+    }
+
+    // Remove event handlers
+    this._events.off(document, this._currentEventType.move, this._onDrag);
+    this._events.off(document, this._currentEventType.stop, this._onDragStop);
+  };
+
+  /**
+   * Get the control position based off the event that fired
+   * @param event - The event to get offsets from
+   */
+  private _getControlPosition(event: MouseTouchEvent<HTMLElement>): ICoordinates | undefined {
+    const touchObj = this._getActiveTouch(event);
+
+    // did we get the right touch?
+    if (this._touchId !== undefined && !touchObj) {
+      return undefined;
+    }
+
+    const eventToGetOffset = touchObj || event;
+    return {
+      x: eventToGetOffset.clientX + document.body.scrollLeft,
+      y: eventToGetOffset.clientY + document.body.scrollTop
```

Comment:
Reading `scrollLeft` and `scrollTop` will cause browser reflow, https://gist.github.com/paulirish/5d52fb081b3570c81e3a#scroll-stuff. Perhaps worth caching / optimizing for?

I'd have to look at the profiler on this branch. I imagine the values don't change often.